server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala

```diff
@@ -105,303 +105,305 @@ class AddressIndexRepository @Inject()(
     */
   def generateQueryAddressRequest(tokens: Map[String, String], originalInput: String): SearchDefinition = {
 
+    val defaultFuzziness = "1"
+
     val saoQuery = Seq(
       tokens.get(Tokens.saoStartNumber).map(token =>
-        matchQuery(
+        constantScoreQuery(matchQuery(
           field = "lpi.saoStartNumber",
           value = token
-        ).boost(queryParams.subBuildingName.lpiSaoStartNumberBoost)),
+        )).boost(queryParams.subBuildingName.lpiSaoStartNumberBoost)),
       tokens.get(Tokens.saoStartSuffix).map(token =>
-        matchQuery(
+        constantScoreQuery(matchQuery(
           field = "lpi.saoStartSuffix",
           value = token
-        ).boost(queryParams.subBuildingName.lpiSaoStartSuffixBoost)),
+        )).boost(queryParams.subBuildingName.lpiSaoStartSuffixBoost)),
       tokens.get(Tokens.saoEndNumber).map(token =>
-        matchQuery(
+        constantScoreQuery(matchQuery(
           field = "lpi.saoEndNumber",
           value = token
-        ).boost(queryParams.subBuildingName.lpiSaoEndNumberBoost)),
+        )).boost(queryParams.subBuildingName.lpiSaoEndNumberBoost)),
       tokens.get(Tokens.saoEndSuffix).map(token =>
-        matchQuery(
+        constantScoreQuery(matchQuery(
           field = "lpi.saoEndSuffix",
           value = token
-        ).boost(queryParams.subBuildingName.lpiSaoEndSuffixBoost))
+        )).boost(queryParams.subBuildingName.lpiSaoEndSuffixBoost))
     ).flatten
 
     val subBuildingNameQuery = Seq(
       tokens.get(Tokens.subBuildingName).map(token =>
-        matchQuery(
+        constantScoreQuery(matchQuery(
           field = "paf.subBuildingName",
           value = token
-        ).boost(queryParams.subBuildingName.pafSubBuildingNameBoost)),
+        )).boost(queryParams.subBuildingName.pafSubBuildingNameBoost)),
       tokens.get(Tokens.subBuildingName).map(token =>
-        matchQuery(
+        constantScoreQuery(matchQuery(
           field = "lpi.saoText",
           value = token
-        ).boost(queryParams.subBuildingName.lpiSaoTextBoost)),
+        )).boost(queryParams.subBuildingName.lpiSaoTextBoost)),
       tokens.get(Tokens.saoStartSuffix).map(token =>
-        matchQuery(
+        constantScoreQuery(matchQuery(
           field = "lpi.saoStartSuffix",
```

Comment:
I am not sure I understand this here. if we have subBuildingName='FLAT 1', then SAO_START_NUMBER should be 1 without suffix and covered by saoQuery. If we have subBuildingNAme='120A-122B', then we need SAO_START_NUMBER=120, SAO_START_SUFFIX=A, SAO_END_NUMBER=122, and SAO_END_SUFFIX=B, which again should be in the saoQuery and not needed here. Have I misunderstood something? Maybe these two need to be joined?